sway/commands.c

```diff
@@ -73,44 +76,55 @@ bool cmd_bindsym(struct sway_config *config, int argc, char **argv) {
 	return true;
 }
 
-bool cmd_exec(struct sway_config *config, int argc, char **argv) {
+static void cmd_exec_cleanup(int signal) {
+	while (waitpid((pid_t)-1, 0, WNOHANG) > 0){};
+}
+
+static bool cmd_exec_always(struct sway_config *config, int argc, char **argv) {
+	/* setup signal handler to cleanup dead proccesses */
+	/* TODO: replace this with a function that has constructor attribute? */
+	static bool cleanup = false;
+	if(cleanup == false) {
+		signal(SIGCHLD, cmd_exec_cleanup);
+		cleanup = true;
+	}
+
 	if (argc < 1) {
 		sway_log(L_ERROR, "Invalid exec command (expected at least 1 argument, got %d)", argc);
 		return false;
 	}
```

Comment:
This might need to go in both, or its own `cmd_exec_checkargs` function.
with this change, invalid `exec` would not show this error during reload which might be unwanted behaviour.